### PR TITLE
Remove dead cache

### PR DESCRIPTION
**Pull request by terencechain**
Part of plan to standardize and improve caching for prysm. This PR removes the dead caches which we'll never use again.

- [x] Removed `active_balance`, `seed`, `start_shard` and `total_balance` caches and their tests
- [x] Fixed existing tests
- [x] Tested run time. 1 node + 64 validators

Reference design: https://docs.google.com/presentation/d/1I9ToaccGX-arYoojzexpj3x22qTsoxsnTT8yUyH-kHY/edit#slide=id.g63b680801c_0_0

### CODE REVIEW

File: validator/accounts/account.go [hunk 1]
```diff
@@ -86,24 +86,18 @@ func NewValidatorAccount(directory string, password string) error {
...
+		return errors.Wrap(err, "could not serialize deposit data")
 	}
-	testAcc.TxOpts.GasLimit = 1000000
-
-	tx, err := testAcc.Contract.Deposit(testAcc.TxOpts, data.PublicKey, data.WithdrawalCredentials, data.Signature)
-	if err != nil {
-		return errors.Wrap(err, "unable to create deposit transaction")
-	}
-	log.Info(`Account creation complete! Copy and paste the raw transaction data shown below when issuing a transaction into the ETH1.0 deposit contract to activate your validator client`)
+	log.Info(`Account creation complete! Copy and paste the deposit data shown below when issuing a transaction into the ETH1.0 deposit contract to activate your validator client`)
```
Comment (0xKiwi) [inline validator/accounts/account.go]: What are these changes?

Comment (terencechain) [reply validator/accounts/account.go]: Bad merge. Thanks for catching this!

File: beacon-chain/cache/feature_flag_test.go [hunk 2]
```diff
@@ -7,9 +7,6 @@ func init() {
 		EnableAttestationCache:   true,
 		EnableActiveBalanceCache: true,
```
Comment (0xKiwi) [inline beacon-chain/cache/feature_flag_test.go]: Does this flag need to be removed as well?

File: shared/featureconfig/config.go [hunk 3]
```diff
@@ -34,9 +34,6 @@ type FeatureFlagConfig struct {
 	EnableAttestationCache   bool // EnableAttestationCache; see https://github.com/prysmaticlabs/prysm/issues/3106.
```
Comment (0xKiwi) [inline shared/featureconfig/config.go]: You should remove what is no longer relevant from here also

Comment (terencechain) [reply shared/featureconfig/config.go]: Done

**Pull request merged**